ui-tests/src/test/java/org/openmrs/reference/EditDemographicTest.java

```diff
@@ -1,75 +1,62 @@
+import org.openmrs.reference.helper.PatientGenerator;
 import org.openmrs.reference.helper.TestPatient;
-import org.openmrs.reference.page.HeaderPage;
-import org.openmrs.reference.page.HomePage;
 import org.openmrs.reference.page.ClinicianFacingPatientDashboardPage;
-import org.openmrs.reference.page.RegistrationPage;
-import org.openmrs.uitestframework.test.TestBase;
-import org.openqa.selenium.By;
-import org.openqa.selenium.support.ui.Select;
+import org.openmrs.reference.page.EditDemographicsPage;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
 
-public class EditDemographicTest extends TestBase {
-    private StringBuffer verificationErrors = new StringBuffer();
-    private HomePage homePage;
-    private HeaderPage headerPage;
-    private RegistrationPage registrationPage;
+public class EditDemographicTest extends ReferenceApplicationTestBase {
+    private EditDemographicsPage editDemographicsPage;
     private ClinicianFacingPatientDashboardPage patientDashboardPage;
 
+    @Test
+    @Category(BuildTests.class)
+    public void editDemographicTest() throws Exception {
+        TestPatient patient = PatientGenerator.generateTestPatient();
+
+        patientDashboardPage = homePage.goToActiveVisitPatient();
+        editDemographicsPage = patientDashboardPage.clickOnEditPatient();
 
-    @Before
-    public void setUp() throws Exception {
+        editDemographicsPage.enterPatientGivenName(patient.givenName);
+        editDemographicsPage.enterPatientMiddleName(patient.middleName);
+        editDemographicsPage.enterPatientFamilyName(patient.familyName);
+        editDemographicsPage.clickOnGenderLabel();
+        editDemographicsPage.selectPatientGender(patient.gender);
+        editDemographicsPage.clickOnBirthdateLabel();
+        editDemographicsPage.enterBirthDay(patient.birthDay);
+        editDemographicsPage.selectBirthMonth(patient.birthMonth);
+        editDemographicsPage.enterBirthYear(patient.birthYear);
+        editDemographicsPage.clickOnConfirmEdit();
+        editDemographicsPage.confirmPatient();
 
-        
-        homePage = new HomePage(page);
-        assertPage(homePage);
-        headerPage = new HeaderPage(driver);
-        registrationPage = new RegistrationPage(driver);
-        patientDashboardPage = new ClinicianFacingPatientDashboardPage(page);
+        String[] names = patientDashboardPage.name().split(" ");
 
-    }
+        assertEquals(names[0], patient.givenName);
+        assertEquals(names[names.length - 1], patient.familyName);
 
-    @Test
-    public void editDemographicTest() throws Exception {
-        homePage.goToActiveVisitPatient();
-        patientDashboardPage.clickOnEditPatient();
-        registrationPage.clearName();
-        registrationPage.enterPatientGivenName("John");
-        registrationPage.clearMiddleName();
-        registrationPage.enterPatientMiddleName("Bob");
```

Comment:
Hi. Sorry for the wait.

The diff looks rather weird, but basically, instead of renaming John Bob Smith to John Bob Smith every test, I decided to generate a random patient each time. The code snippet above is directly replaced by
```Java
        TestPatient patient = PatientGenerator.generateTestPatient();

        patientDashboardPage = homePage.goToActiveVisitPatient();
        editDemographicsPage = patientDashboardPage.clickOnEditPatient();

        editDemographicsPage.enterPatientGivenName(patient.givenName);
        editDemographicsPage.enterPatientMiddleName(patient.middleName);
        editDemographicsPage.enterPatientFamilyName(patient.familyName);
```